plugins/callback/diy.py

```diff
@@ -0,0 +1,1421 @@
+
+      - name: indentation
+        debug:
+          msg: "{{ item.msg }}"
+        with_items:
+          - { indent: 1, msg: one., color: red }
+          - { indent: 2, msg: two.., color: yellow }
+          - { indent: 3, msg: three..., color: bright yellow }
+        vars:
+          ansible_callback_diy_runner_item_on_ok_msg: "{{ ansible_callback_diy.result.output.msg | indent(item.indent, True) }}"
+          ansible_callback_diy_runner_item_on_ok_msg_color: "{{ item.color }}"
+          ansible_callback_diy_runner_on_ok_msg: "GO!!!"
+          ansible_callback_diy_runner_on_ok_msg_color: bright green
+
+      - name: using lookup and template as file
+        shell: "echo {% raw %}'output from {{ file_name }}'{% endraw %} > {{ file_name }}"
+        vars:
+          ansible_callback_diy_playbook_on_task_start_msg: "\ndiy output(via task vars): task example: {{ ansible_callback_diy.task.name }}"
+          file_name: diy_file_template_example
+          ansible_callback_diy_runner_on_ok_msg: "{{ lookup('template', file_name) }}"
+
+      - name: 'look at top level vars available to the "runner_on_ok" callback'
+        debug:
+          msg: ''
+        vars:
+          ansible_callback_diy_playbook_on_task_start_msg: "\ndiy output(via task vars): task example: {{ ansible_callback_diy.task.name }}"
+          ansible_callback_diy_runner_on_ok_msg: |+2
+            {% for var in (ansible_callback_diy.top_level_var_names|reject('match','vars|ansible_callback_diy.*')) | sort %}
+            {{ green }}{{ var }}:
+              {{ white }}{{ lookup('vars', var) }}
+
+            {% endfor %}
+          ansible_callback_diy_runner_on_ok_msg_color: white
+
+      - name: 'look at event data available to the "runner_on_ok" callback'
+        debug:
+          msg: ''
+        vars:
+          ansible_callback_diy_playbook_on_task_start_msg: "\ndiy output(via task vars): task example: {{ ansible_callback_diy.task.name }}"
+          ansible_callback_diy_runner_on_ok_msg: |+2
+            {% for key in ansible_callback_diy | sort %}
+            {{ green }}{{ key }}:
+              {{ white }}{{ ansible_callback_diy[key] }}
+
+            {% endfor %}
+'''
+
+import sys
+from contextlib import contextmanager
+from ansible import constants as C
+from ansible.playbook.task_include import TaskInclude
+from ansible.plugins.callback import CallbackBase
+from ansible.utils.color import colorize, hostcolor
+from ansible.template import Templar
+from ansible.vars.manager import VariableManager
+from ansible.plugins.callback.default import CallbackModule as Default
+from ansible.module_utils._text import to_text
+
+
+class CallbackModule(Default):
+    """
+    Callback plugin for the DIYer, because it allows you to supply your own custom callback templates to be output
+    """
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'stdout'
+    CALLBACK_NAME = 'diy'
+
+    DIY_NS = 'ansible_callback_diy'
+
+    class DummyStdout(object):
```

Comment:
Why not move this one level up?

Reply:
@felixfontein Just want to get clarification on your suggestion here. I'm not sure if you're suggesting to just move the `DummyStdout` class up to the base class? Or if you're suggesting to move the `DummyStdout` logic's intent up to the base class (e.g. add a control that would prevent output)? Or perhaps something else entirely, and I'm completely misunderstanding 😄

Reply:
No, put `DummyStdout` to the top level of this file. Inner classes are not something that is common in Python, and so this is confusing more people looking at your code than helping.

Reply:
Understood. Thank you for the clarification. I made the change.